src/pages-and-resources/data/slice.js

```diff
@@ -23,6 +25,12 @@ const slice = createSlice({
     updateCourseAppsApiStatus: (state, { payload }) => {
       state.courseAppsApiStatus = payload.status;
     },
+    fetchCourseAppsSettingsSuccess: (state, { payload }) => {
+      Object.assign(state.courseAppSettings, payload);
+    },
+    updateCourseAppsSettingsSuccess: (state, { payload }) => {
+      Object.assign(state.courseAppSettings, payload);
```

Comment:
any reason why we are using object.assign instead of directly assigning the value? just curious because I think the create slice manages reference updates itself.

Reply:
`courseAppSettings` will store all settings across all apps, while the payload will only return settings for a single app. So if we directly assign, then fetching one setting will clear the others.